lib/broadcast-operator.ts

```diff
@@ -258,19 +258,22 @@ interface SocketDetails {
  */
 export class RemoteSocket<EmitEvents extends EventsMap>
   implements TypedEventBroadcaster<EmitEvents> {
+  public readonly adapter: Adapter;
   public readonly id: SocketId;
   public readonly handshake: Handshake;
   public readonly rooms: Set<Room>;
   public readonly data: any;
 
   private readonly operator: BroadcastOperator<EmitEvents>;
+  private flags: BroadcastFlags = {};
 
   constructor(adapter: Adapter, details: SocketDetails) {
+    this.adapter = adapter;
     this.id = details.id;
     this.handshake = details.handshake;
     this.rooms = new Set(details.rooms);
     this.data = details.data;
-    this.operator = new BroadcastOperator(adapter, new Set([this.id]));
+    this.operator = new BroadcastOperator(adapter, new Set(details.rooms));
```

Comment:
That does not seem correct. Could you please check?

Reply:
@darrachequesne Initially I thought that having the rooms set to the RemoteSocket rooms is going to be more logical, but now I see Socket has the same thing. I'll revert it.